File: 1518-water-bottles/1518-water-bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        res = 0 
        bottles = 0
        while True :
            if numBottles == 0 and bottles < numExchange : break
            res+=numBottles
            bottles+=numBottles
            n = bottles//numExchange
            numBottles = n
            bottles-=n*numExchange
        return res

File: 1518-water-bottles/1518-water-bottles.cpp
class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        int empty = 0;
        int res = 0;
        while(numBottles!=0){
            res+=numBottles;
            empty+=numBottles;
            numBottles = 0;
            numBottles+= empty/numExchange;
            empty-=numExchange * numBottles;
        }
        return res;
    }
};

